ArcGISRuntimeSDKQt_CppSamples/Scenes/OrbitCameraAroundObject/OrbitCameraAroundObject.cpp

```diff
@@ -0,0 +1,282 @@
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "SimpleRenderer.h"
+
+#include <QDir>
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+  // helper method to get cross platform data path
+  QString defaultDataPath()
+  {
+    QString dataPath;
+
+    #ifdef Q_OS_ANDROID
+      dataPath = "/sdcard";
+    #elif defined Q_OS_IOS
+      dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+    #else
+      dataPath = QDir::homePath();
+    #endif
+
+    return dataPath;
+  }
+}
+
+OrbitCameraAroundObject::OrbitCameraAroundObject(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(Basemap::imagery(this), this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+}
+
+OrbitCameraAroundObject::~OrbitCameraAroundObject() = default;
+
+void OrbitCameraAroundObject::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<OrbitCameraAroundObject>("Esri.Samples", 1, 0, "OrbitCameraAroundObjectSample");
+}
+
+SceneQuickView* OrbitCameraAroundObject::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void OrbitCameraAroundObject::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+  {
+    return;
+  }
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  /* Scene Rendering setup */
+  // create a new graphics overlay and add it to the sceneview
+  GraphicsOverlay* sceneOverlay = new GraphicsOverlay(this);
+  sceneOverlay->setSceneProperties(LayerSceneProperties(SurfacePlacement::Relative));
+  m_sceneView->graphicsOverlays()->append(sceneOverlay);
+
+  // create a renderer to control the plane's orientation by its attributes
+  SimpleRenderer* renderer3D = new SimpleRenderer(this);
+  {
+    //Set the renderer pitch/heading expressions, so the plane can be oriented via properties.
+    RendererSceneProperties properties = renderer3D->sceneProperties();
+    properties.setPitchExpression("[PITCH]");
+    properties.setHeadingExpression("[HEADING]");
+    renderer3D->setSceneProperties(properties);
+  }
+  sceneOverlay->setRenderer(renderer3D);
+
+
+  /* Plane model Setup */
+  //Load a plane model.
+  QUrl planeFileURl = QUrl(defaultDataPath() + "/ArcGIS/Runtime/Data/3D/Bristol/Collada/Bristol.dae");
+  ModelSceneSymbol* planeModel = new ModelSceneSymbol(planeFileURl, 1.0, this);
+  planeModel->load();
+
+  //Create a graphic from the plane model, positioned atop a runway.
+  Point runwayPos = Point{6.637, 45.399, 100, SpatialReference::wgs84()};
```

Comment:
Our minimum supported Linux compiler is 4.8. As such - do check that initializer braces work as it is temperamental. I've checked for you and can confirm it compiles on Redhat with 4.8.

Reply:
Ach, that's scary.
D'you recommend defaulting against braced initialization in general?